x/orderbook/controller.go

```diff
@@ -0,0 +1,166 @@
+package orderbook
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/x/cash"
+)
+
+func orderCondition(id []byte) weave.Condition {
+	if len(id) == 0 {
+		panic("developer error: must save before taking address")
+	}
+	return weave.NewCondition("orderbook", "order", id)
+}
+
+// Address returns unique address, so an order contract can manage funds
+func (order *Order) Address() weave.Address {
+	return orderCondition(order.ID).Address()
+}
+
+type controller struct {
+	orderBucket *OrderBucket
+	tradeBucket *TradeBucket
+	mover       cash.CoinMover
+}
+
+func (c controller) settleOrder(db weave.KVStore, order *Order, now weave.UnixTime) error {
+	var other Side
+	var descending bool
+	var acceptable func(our, their *Amount) bool
+
+	// We configure which orders to search and which direction
+	if order.Side == Side_Ask {
+		// we want the highest price for our Ask token
+		other = Side_Bid
+		descending = true
+		acceptable = func(our, their *Amount) bool { return !our.Greater(their) }
+	} else {
```

Comment:
[SIDE_INVALID](https://github.com/iov-one/tutorial/blob/master/x/orderbook/codec.proto#L38...L42) you see at ```codec.proto``` is enforced by prototool. It is not included it in business logic therefore it is validated inside [model.go](https://github.com/iov-one/tutorial/blob/master/x/orderbook/model.go#L136). But if you say we should make an extra check with ```switch```, why not?

Reply:
We could use a boolean instead.

It is an enum that will only ever have two values. Invalid is enforced by our lovely linter. I find it nicer doing `if order.Side == Side_Bid` rather than `if !order.Ask` but we could do the second to make it clear there will only ever be two values

Reply:
Changed side to IsAsk bool in 5092fa5